src/java/org/apache/cassandra/tools/FieldUtil.java

```diff
@@ -27,19 +28,44 @@ public static void setInstanceUnsafe(Class<?> klass, Object v, String fieldName)
     {
         try
         {
-            Field field = klass.getDeclaredField(fieldName);
-            field.setAccessible(true);
+            setInstanceUnsafeThrowing(klass, v, fieldName);
+        }
+        catch (Throwable e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
 
-            Field modifiersField = Field.class.getDeclaredField("modifiers");
-            modifiersField.setAccessible(true);
-            modifiersField.setInt(field, field.getModifiers() & ~Modifier.FINAL);
+    private static void setInstanceUnsafeThrowing(Class<?> klass, Object v, String fieldName) throws Throwable
+    {
+        Field field = klass.getDeclaredField(fieldName);
+        field.setAccessible(true);
 
-            field.set(null, v);
+        try
+        {
+            Field modifiers = Field.class.getDeclaredField("modifiers");
```

Comment:
this pattern was fixed awhile back, can we use `org.apache.cassandra.utils.ReflectionUtils#getModifiersField` rather than do our own thing?